src/api/localStorage.js

```diff
@@ -0,0 +1,96 @@
+export class LocalStorage {
+  constructor() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  createUser(user) {
+    this.updateUsers();
+    user.id = this.generateId();
+    this.users.push(user);
+    this.setLocalStorageData();
+  }
+
+  setLocalStorageData() {
+    localStorage.setItem('UserAccounts', JSON.stringify(this.users));
+  }
+
+  updateUsers() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  getUser(id) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    return user;
+  }
+
+  getAllUsers() {
+    return this.users;
+  }
+
+  editUser(id, editName, editValue) {
+    try{
+      const editObj = {};
+      editObj[`${editName}`] = `${editValue}`;
+      const updatedUser = Object.assign(this.getUser(id), editObj);
+      this.deleteUser(id);
```

Comment:
Although this is pretty clever
```
      this.deleteUser(id);
      this.createUser(updatedUser);
```
you could not only replace the user with a single line, but you wouldn't have to set `localStoarge` twice for a single action

```suggestion
      this.users.splice(this.users.findIndex(user => user.id === id), 1, updatedUser);
      this.syncStorage();
```

Reply:
Not sure if this one will work since I am using createUser() to generate my user id. Are you saying to remove the calls to deleteUser() and createUser() or just deleteUser()? Confused cause I see we're passing in the updated user to the splice.

Reply:
why do you need to generate an id if you're only editing a user?

Reply:
> Are you saying to remove the calls to deleteUser() and createUser() or just deleteUser()?

remove both

Reply:
I see so the updatedUser will keep the current users id. Got it, thank you.